components/style/stylist.rs

```diff
@@ -1677,11 +1673,7 @@ impl PerOriginCascadeData {
     }
 
     fn clear(&mut self) {
-        self.element_map = SelectorMap::new();
-        self.pseudos_map = Default::default();
-        SelectorImpl::each_eagerly_cascaded_pseudo_element(|pseudo| {
-            self.pseudos_map.insert(pseudo, SelectorMap::new());
-        });
+        *self = Self::new();
```

Comment:
Hmm... Good call, perhaps! I'll measure and see if there's any difference, didn't know `clear()` reused heap allocs.